[alvinnzz] iP

alvinnzz opened the pull request:
## DukeBot
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukeBot frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

alvinnzz committed:
Created greet and goodbye functions

alvinnzz committed:
Created echo function to echo user input and call goodbye function

alvinnzz committed:
Created function to add task to list and print list

alvinnzz committed:
Created Task class and added ability to mark tasks as done or not done

alvinnzz committed:
Created subclasses Deadline, Event, Todo for Task and functions edited

alvinnzz committed:
Changed input and expected text and tested with automated text UI testing

alvinnzz committed:
Created DukeException class to handle exceptions for function addToList

alvinnzz committed:
Created delete function and changed the use of Array to ArrayList

alvinnzz committed:
Created level 7 branch and added Storage class which has functions save and load for tasklist

alvinnzz committed:
Merge branch 'branch-Level-7'

alvinnzz committed:
Updated tasks to store date and print the date for the tasks.

alvinnzz committed:
Merge branch 'branch-level-8'

alvinnzz committed:
Added Duke.Command package and several commands class.

alvinnzz committed:
Added Duke and Duke.Task package and created classes Parser, Ui

alvinnzz committed:
Created 3 more commands for the 3 different tasks.

alvinnzz committed:
Merge branch 'add-gradle-support' of https://github.com/alvinnzz/ip

alvinnzz committed:
Created test cases using JUnit for TaskList and Storage class.

alvinnzz committed:
Released jar file for the chatbot and removed unused imports.

alvinnzz committed:
Update javadoc for Duke.Tasks package

alvinnzz committed:
Update javadoc for Duke package.

alvinnzz committed:
Update javadoc for Duke.Commands package

alvinnzz committed:
Update files to meet coding standard

alvinnzz committed:
Add command to find tasks by keyword

alvinnzz committed:
Merge branch 'branch-A-JavaDoc'

alvinnzz committed:
Merge branch 'branch-A-CodingStandard' and resolve conflicts

alvinnzz committed:
Merge branch 'branch-Level-9' and resolve conflicts

alvinnzz committed:
Add config files to project

alvinnzz committed:
Fix style for Duke Commands package

alvinnzz committed:
Merge branch 'branch-A-CheckStyle'

Vision-2000 commented on src/main/java/Duke/Commands/CreateEventCommand.java:
```diff
@@ -0,0 +1,55 @@
+package Duke.Commands;
```
Should the package names be in lower case?

Vision-2000 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+package Duke;
+
+import Duke.Commands.*;
```
Should the imported classes be listed out?

Vision-2000 commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Prints goodbye message to the user.
+     */
+    public void sayGoodbye() {
+        String exitMessage = "____________________________________________________________\n" +
+                " Bye. Hope to see you again soon!\n" +
+                "____________________________________________________________\n";
+        System.out.println(exitMessage);
+    }
```
I like the distinction in names between methods that are called once vs methods that can be called multiple times.

Vision-2000 commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+    public static Command parse(String input, Ui ui, TaskList taskList, Storage storage) throws DukeException {
+        try {
+            String[] inputs = input.split(" ");
+            CommandWords commandWord = CommandWords.valueOf(inputs[0].toUpperCase());
```
I like that the variable here is correctly named in singular form to indicate that it is a word, in contrast with the class name which is plural.

Vision-2000 reviewed (commented):
LGTM. There were very little nits and code was well organised.

alvinnzz committed:
Rename directories and packages

alvinnzz committed:
Change Ui to return String to prepare for GUI

alvinnzz committed:
Create GUI for Duke chatbot

alvinnzz committed:
Merge branch 'branch-Level-10' and merge conflicts

RingoftheKing commented on src/main/java/duke/commands/CreateEventCommand.java:
```diff
@@ -0,0 +1,55 @@
+        String[] inputArr = input.split(" ");
+        String commandWord = inputArr[0];
+        try {
+            addedTask = new Event(getTaskName(commandWord, this.input),
```
nitpick: this keyword may not be necessary

RingoftheKing commented on src/main/java/duke/commands/CreateEventCommand.java:
```diff
@@ -0,0 +1,55 @@
+            taskList.addTask(addedTask);
+            return this.ui.printAddMessage(taskList.size(), addedTask);
+        } catch (DateTimeException dte) {
+            return this.ui.printDateErrorMessage("Please follow the format for date and time (d/M/yyyy HHmm).");
```
For the user, it may be hard to understand what `d/M/yyyy HHmm` means without knowing Date Formatters of Java. Perhaps print a more user-friendly advice?

RingoftheKing commented on src/main/java/duke/commands/ErrorCommand.java:
```diff
@@ -0,0 +1,39 @@
+import duke.Ui;
+
+/**
+ * Represents a command to print an error.
```
It would be nice to indicate when this class will be used/reached/activated.

RingoftheKing commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+     *
+     * @param index Index of task to be set as done.
+     */
+    public void setAsDone(int index) {
+        this.taskList.get(index).setAsDone();
+    }
+
+    /**
+     * Set task as not done in the TaskList.
+     *
+     * @param index Index of task to be set as not done.
+     */
+    public void setAsNotDone(int index) {
+        this.taskList.get(index).setAsNotDone();
```
Great use of abstraction here!

RingoftheKing reviewed (commented):
LGTM, your code has its concerns very well separated into its required packages and classes

alvinnzz committed:
Add assertions to enhanced code reliability

Assertions are added to help debugging, validate assumptions and detect unexpected conditions during runtime.

alvinnzz committed:
Merge pull request #1 from alvinnzz/branch-A-Assertions

Add assertions to code.

alvinnzz committed:
Merge branch 'master' of https://github.com/alvinnzz/ip

alvinnzz committed:
Improve code quality

Improve code readability and ensure that happy path is prominent with appropriate comments

alvinnzz committed:
Merge branch 'master' into branch-A-CodeQuality

alvinnzz committed:
Merge pull request #2 from alvinnzz/branch-A-CodeQuality

Improve code quality

alvinnzz committed:
Add reminder command to remind user about upcoming deadlines

alvinnzz committed:
Merge branch 'branch-B-Reminders'

alvinnzz committed:
Fix style issues

alvinnzz committed:
Fix getDate function in Deadline and Event class

alvinnzz committed:
Include sayHello in chatBot GUI

alvinnzz committed:
Fix style and removed break lines in Ui message

alvinnzz committed:
Update user guide and include ui picture

alvinnzz committed:
Update the gradle build main application to Launcher
